doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+tools such as ``bash``, ``make`` or the package manager ``pacman``.
+
+MSYS2 comes with different environments. You can learn more about the different
+environments in the `MSYS2 documentation
+<https://www.msys2.org/docs/environments/>`_. We will be using ``UCRT64`` in the
+examples below but it can be swapped by ``CLANG64`` (as long as the external
+packages you want to use are available for the ``CLANG64`` environment, see
+:ref:`Using system libraries with GHC`).
+
+.. note::
+
+    When the code snippets must be run in a MSYS2 shell, we will use the ``msys2>``
+    prompt and when in a native Windows shell we will use the ``PS>`` or ``cmd>``
+    prompt. If either shell will do, we will just use ``>``.
+
+A small note on configuring and updating MSYS2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before starting with MSYS2, one should make sure it is updated and properly configured. For that:
+
+::
+
+   # Update the system, run twice
+   msys2> pacman --noconfirm -Syuu
+   msys2> pacman --noconfirm -Syuu
+
+   # install basic utilities
+   msys2> pacman --noconfirm -S --needed curl autoconf mingw-w64-ucrt-x86_64-pkgconf
+   msys2> pacman --noconfirm -S ca-certificates
+
+   # Optionally if you like zsh
+   msys2> pacman --noconfirm -S zsh
+
+   # make $HOME the same in both PowerShell and MSYS2's Bash
+   msys2> sed -i -e 's/db_home:.*$/db_home: windows/' /etc/nsswitch.conf
+
+   # Inherit Windows' path in MSYS2
+   msys2> sed -i -e 's/rem set MSYS2_PATH_TYPE=inherit/set MSYS2_PATH_TYPE=inherit/' /c/msys64/msys2_shell.cmd
+
+As a terminal, `Windows Terminal
+<https://apps.microsoft.com/detail/9N0DX20HK701>`_ usually comes in handy. I use
+the following profile:
+
+::
+
+   {
+     "colorScheme": "One Half Dark",
+     "commandline": "C:\\msys64\\msys2_shell.cmd -defterm -here -no-start -ucrt64 -shell zsh",
+     "hidden": false,
+     "icon": "C:\\msys64\\ucrt64.ico",
+     "name": "MSYS2 / UCRT64",
+     "startingDirectory": "C:/Users/<YourName>"
+   }
+
+Installing a Haskell environment on Windows
+-------------------------------------------
+
+The recommended way of installing Haskell in Windows is via `GHCup
+<https://www.haskell.org/ghcup/>`_. There are however two different ways of
+using GHCup on Windows:
+
+GHCup without having installed MSYS2 already
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When the user just has fresh system, this option will install MSYS2 as a first
+step. The command, as presented in the front page of `GHCup's website
+<https://www.haskell.org/ghcup/>`_, has to be executed on a PowerShell:
+
+::
+
+   PS> Set-ExecutionPolicy Bypass -Scope Process -Force;[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; try { Invoke-Command -ScriptBlock ([ScriptBlock]::Create((Invoke-WebRequest https://www.haskell.org/ghcup/sh/bootstrap-haskell.ps1 -UseBasicParsing))) -ArgumentList $true } catch { Write-Error $_ }

```

Comment:
As above... I don't think cabal should provide this type of information. It is likely going to be out of date and cause issues when I change the default arguments etc.

Reply:
Agree. This section is now gone.